src/utils/transformation.ts

```diff
@@ -0,0 +1,53 @@
+import { BadRequestException } from '@nestjs/common';
+import { TransformFnParams } from 'class-transformer';
+
+export class Transformation<T> {
+  constructor(
+    private readonly transformFunction: (value: any) => T,
+    private readonly nullable = true,
+    private readonly protoName = '',
```

Comment:
why `protoName` is empty string, does it need to be type that u expect them to put in?

Reply:
protoName is a optional parameter to show expected type that dto want. Default value is empty string but you can take a name of the type to parameter.

Reply:
gotcha